src/classes/providers/base/types.ts

```diff
@@ -18,8 +18,8 @@ interface ICreateProxyConfig {
 interface IProviderConfig {
   host: string;
   port: number;
-  username: string;
-  password: string;
+  username?: string;
+  password?: string;
```

Comment:
Are you sure that's the best way to put these attributes as optional for all proxy providers ?
Isn't it better to create you own "Shifter" interface ?